pydra/utils/tests/test_typing.py

```diff
@@ -164,6 +167,18 @@ def test_type_check_nested8():
         )(lz(ty.List[float]))
 
 
+def test_type_check_permit_superclass():
+    # Typical case as Json is subclass of File
+    TypeParser(ty.List[File])(lz(ty.List[Json]))
```

Comment:
should we give some warning?

Reply:
It is logged as an "info" rather than a warning, since I figured that there is no harm in relying on it